modules/oe_content_call_proposals/tests/src/Kernel/CallForProposalsNodeWrapperTest.php

```diff
@@ -0,0 +1,399 @@
+   */
+  public function testUnsupportedEntityType(): void {
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage("The current wrapper only accepts 'node' entities.");
+    $entity = EntityTest::create();
+    CallForProposalsNodeWrapper::getInstance($entity);
+  }
+
+  /**
+   * Test exception thrown if an unsupported entity bundle is passed.
+   */
+  public function testUnsupportedEntityBundle(): void {
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage("The current wrapper only accepts 'node' entities of type 'oe_call_proposals'.");
+    $node_type = NodeType::create([
+      'type' => 'test',
+      'name' => 'Test bundle',
+    ]);
+    $node_type->save();
+    $entity = Node::create([
+      'type' => 'test',
+    ]);
+    CallForProposalsNodeWrapper::getInstance($entity);
+  }
+
+  /**
+   * Test getter methods.
+   */
+  public function testGetters(): void {
+    foreach ($this->getterAssertions() as $assertion) {
+      $this->assertGetters($assertion['case'], $assertion['values'], $assertion['assertions']);
+    }
+  }
+
+  /**
+   * Assert getter methods.
+   *
+   * @param string $case
+   *   Test case description.
+   * @param array $values
+   *   Entity values to initialize the wrapper with.
+   * @param array $assertions
+   *   Assertions to be ran over it.
+   */
+  protected function assertGetters(string $case, array $values, array $assertions): void {
+    // Create wrapper.
+    $wrapper = $this->createWrapper($values);
+
+    // Run assertions.
+    foreach ($assertions as $method => $expected) {
+      $actual = $wrapper->{$method}();
+      if ($actual instanceof DrupalDateTime) {
+        $actual = $actual->format(DrupalDateTime::FORMAT);
+      }
+      $this->assertEquals($expected, $actual, "Test case '{$case}' failed: method {$method} did not return what was expected.");
+    }
+  }
+
+  /**
+   * Create a wrapper object given its node entity values.
+   *
+   * @param array $values
+   *   Entity values.
+   *
+   * @return \Drupal\oe_content_call_proposals\CallForProposalsNodeWrapper
+   *   Wrapper object.
+   */
+  protected function createWrapper(array $values): CallForProposalsNodeWrapperInterface {
+    // Create wrapper.
+    $node = Node::create($values + [
+      'type' => 'oe_call_proposals',
+      'title' => 'My call for proposal',
+    ]);
+    $node->save();
+    return CallForProposalsNodeWrapper::getInstance($node);
+  }
+
+  /**
+   * Assertion for wrapper getters.
+   *
+   * This would normally be a data provider but it would require a full test
+   * bootstrap for each test case, which will add minutes to test runs.
+   *
+   * Since we are testing a simple entity wrapper we will instead run it in
+   * the same test.
+   *
+   * @return array
+   *   List of assertion for wrapper test.
+   */
+  public function getterAssertions(): array {
+    return [
```

Comment:
Lets add assertions for isDeadlineModelPermanent here as well since we added that method.

Reply:
I added the assertions.